salt/modules/nilrt_ip.py

```diff
@@ -469,10 +584,15 @@ def down(interface, iface_type=None):  # pylint: disable=unused-argument
         salt '*' ip.down interface-label
     '''
     if __grains__['lsb_distrib_id'] == 'nilrt':
-        out = __salt__['cmd.run_all']('ip link set {0} down'.format(interface))
-        if out['retcode'] != 0:
-            msg = 'Couldn\'t disable interface {0}. Error: {1}'.format(interface, out['stderr'])
-            raise salt.exceptions.CommandExecutionError(msg)
+        initial_mode = _get_adapter_mode_info(interface)
```

Comment:
up/down code is almost identical. Can you create a worker to share the logic?

Reply:
A function called `_change_state(state)` and state can be `up` or `down` ?

Reply:
Looks good to me!